.github/workflows/build_and_test_matrix.yml

```diff
@@ -0,0 +1,39 @@
+name: Build and test matrix
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+        python-version: ['3.7', '3.8', '3.9']
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+         pip install --upgrade pip
+         pip install flake8 pytest coverage wheel
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        # flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        flake8 . --count --select=E9,F63,F7 --show-source --statistics
```

Comment:
We could use --select=E9,F63,F7,F82 after fixing the undefined variables that are just remnants of the Python2/3 compatibility code.

Reply:
Sure, I’ll let you do it :)